app/connectors/aws/S3FileManager.scala

```diff
@@ -88,19 +88,25 @@ class S3FileManager @Inject()(s3Client: AmazonS3, clock: Clock)(implicit ec: Exe
     }
 
   override def withObjectContent[T](objectLocation: S3ObjectLocation)(function: ObjectContent => Future[T])(
-    implicit loggingDetails: LoggingDetails): Future[T] =
-    Future {
+    implicit loggingDetails: LoggingDetails): Future[T] = Future {
       val getObjectRequest = new GetObjectRequest(objectLocation.bucket, objectLocation.objectKey)
       objectLocation.objectVersion.foreach(getObjectRequest.setVersionId)
       val fileFromLocation = s3Client.getObject(getObjectRequest)
-      val content =
-        ObjectContent(fileFromLocation.getObjectContent, fileFromLocation.getObjectMetadata.getContentLength)
+      val fileInputStream = fileFromLocation.getObjectContent
+      val content: ObjectContent =
+        ObjectContent(fileInputStream, fileFromLocation.getObjectMetadata.getContentLength)
       withLoggingDetails(loggingDetails) {
         Logger.debug(s"Fetched content for objectKey: [${objectLocation.objectKey}].")
       }
 
       val result: Future[T] = function(content)
-      result.onComplete(_ => fileFromLocation.close())
+      result.onComplete(_ => {
+        // This is required to ensure the full reading of the InputStream as
+        // without reading the InputStream we see large amounts of error messages:
+        // Not all bytes were read from the S3ObjectInputStream
+        Stream.continually(fileInputStream.read).takeWhile(_ != -1)
```

Comment:
In the JIRA you mentioned that

    fileInputStream.abort()

... might work. Did it not?

Reply:
Slightly concerned that this might run forever...

Reply:
Ah my bad. There were two options we could use: ensure the InputStream is fully read, OR abort it. After some reading I thought to try the full reading first, as the stream should be being fully read anyway and aborting it as a bit hacky. I'll do the abort() if fully reading doesn't resolve.